src/http/client.cr

```diff
@@ -58,6 +58,8 @@
 # client.close
 # ```
 #
+# WARNING: The implementation of `HTTP::Client` does not support concurrent requests.
```

Comment:
I think we should start using some terminology for this. I wonder if this could be understood as "don't use two HTTP::Client concurrently".

Maybe something like "A single HTTP::Client instance does not support concurrent requests"

In Go they use the phase "safe for concurrent use by multiple goroutines", so maybe we should use "(un)safe for concurrent use by multiple fibers", something like that.